calico/felix/fetcd.py

```diff
@@ -334,6 +335,57 @@ def __init__(self, config, etcd_api, status_reporter, hosts_ipset):
         self.last_rate_log_time = monotonic_time()
         # Register for events when values change.
         self._register_paths()
+        self._usage_report_greenlet = gevent.Greenlet(
+            self._periodically_usage_report
+        )
+
+    def estimated_host_count(self):
+        """
+        Thread safe for caller because calling len() on a dict is atomic in Python.
+
+        :return An aproximation to our Felix cluster count
+        """
+        return len(self.ipv4_by_hostname)
+
+    @logging_exceptions
+    def _periodically_usage_report(self):
+        """
+        Greenlet: periodically report the cluster existence to
+        projectcalico.org.  Period is about once per day.
+
+        :return: Does not return, unless USAGE_REPORT disabled.
+        """
+
+        interval = 86400   # Once every 24 hours minus 12 minute jitter
+        jitter = random.random() * 0.01 * interval
+        try:
+            calico_version = str(pkg_resources.require("calico")[0].version)
+        except ResolutionError:
+            calico_version = "NA"
+
+        _log.info("Started usage report thread.  Usage report interval: %s, pre-jitter: %s", interval, jitter)
+
+        # Pre-Jitter the reporting thread start by 1% of interval (about 12 minutes)
+        # Jitter prevents thundering herd for large clusters when the cluster first starts
+        # Do pre-jitter only for clusters greater than 25.
+        felix_count = self.estimated_host_count()
+        if (felix_count >= 25):
+            gevent.sleep(jitter)
+
+        while True:
+            config = self._config
+            felix_count = self.estimated_host_count()
+            cluster_type = "NA"
```

Comment:
I don't know why we are including this now?  Can't additional fields be added in the future trivially?